commandline/src/main/java/gov/cms/qpp/conversion/CommandLineRunner.java

```diff
@@ -0,0 +1,214 @@
+package gov.cms.qpp.conversion;
+
+import gov.cms.qpp.conversion.segmentation.QrdaScope;
+import gov.cms.qpp.conversion.util.Finder;
+
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.io.UncheckedIOException;
+import java.nio.file.FileSystem;
+import java.nio.file.FileSystems;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.EnumSet;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+import org.apache.commons.cli.CommandLine;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Responsible for executing the given command line instructions
+ */
+public class CommandLineRunner implements Runnable {
+
+	private static final Logger DEV_LOG = LoggerFactory.getLogger(CommandLineMain.class);
+	private static final Pattern LITERAL_COMMA = Pattern.compile(",", Pattern.LITERAL);
+	private static final Pattern NORMAL_PATH = Pattern.compile("[a-zA-Z0-9_\\-\\s,.\\/]+");
+	private static final Pattern GLOB_FINDER = Pattern.compile("(" + NORMAL_PATH.pattern() + ").+");
+
+	private final CommandLine commandLine;
+	private final FileSystem fileSystem;
+	private Set<QrdaScope> scope;
+	private boolean doValidation;
+	private boolean doDefaults;
+	private boolean historical;
+
+	/**
+	 * Creates a new CommandLineRunner from a given {@link CommandLine}
+	 *
+	 * @param commandLine command line execute
+	 */
+	public CommandLineRunner(CommandLine commandLine) {
+		this(commandLine, FileSystems.getDefault());
+	}
+
+	/**
+	 * Creates a new CommandLineRunner from a given {@link CommandLine} and {@link FileSystem}
+	 *
+	 * @param commandLine command line execute
+	 * @param fileSystem contextual file system to use for path operations
+	 */
+	public CommandLineRunner(CommandLine commandLine, FileSystem fileSystem) {
+		Objects.requireNonNull(commandLine, "commandLine");
+		Objects.requireNonNull(fileSystem, "fileSystem");
+
+		this.commandLine = commandLine;
+		this.fileSystem = fileSystem;
+	}
+
+	/**
+	 * Executes the command line
+	 */
+	@Override
+	public void run() {
```

Comment:
there's a fair amount of nesting here, can we piece this up?

Reply:
Originally it was nice and flat, but sonar got upset about that. I'm open to suggestions on splitting the method up.

Reply:
Hrm... regardless of the nesting, it does seem like a fairly clean and readable function.  I'm not sure whether creating some more helper functions to reduce the nesting in this function will be helpful or if it will create more mess.  I'm inclined to say that if we can't come up with a reasonable refactoring within a short timeframe, let's create a refactor issue in our backlog.  Make sure we reference the function in the issue and why we would like to refactor it.  Tag me on it to remind me to setup a short meeting to discuss it to help with brainstorming.